### DISCUSSION TITLE
Renovate integration with Bitbucket Server

### DISCUSSION THREAD

**Discussion by mikemicky4321** [renovatebot/renovate#18368] [Request Help] [answered]
Hello all,

Firstly thanks for the great product. We are currently using Bitbucket server. I have read all the documentation and found that Self-hosting method we have to use for Bitbucket. We followed the following steps,

1. Created an account in Bitbucket with username renovate-bot. Please find the used config.js and renovate.json below. We are using 
**config.js**

```
module.exports = {
  platform: "bitbucket-server",
  username: "renovate-bot",
  password: "xxxxxxxxx",
  endpoint: "https://example.com/",
  onboarding: false,
  logLevel: "debug",
  dryRun: true,
  autodiscover: true,
};

```
**renovate.json**
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json"
}
```
2. The config.js file was mounted to the docker. We manually using PR integrate renovate.json to the repository. By Running this command `docker run --rm -v "/home/renovate/renovate_config/config.js:/usr/src/app/config.js" renovate/renovate` we can see the scan of all the repositories in the bitbucket with number of files. But there was no PR was created by renovate with Vulnerable libraries.

Could anyone help us to configure renovate?
Thanks for your support in advance

**Comment by viceice** [comment, marked as answer]
log level is an environment option only.

the logs should indicate why there is no PR.

are there branches from renovate?

do you have branch protections which blocking renovate from crating branches?

**Reply by mikemicky4321** [reply]
Hello @viceice ,
Yes, we thought the same i.e, the logs should display about PR. But there was no indication. There are no branches from renovate. We don't have any branch protections. We are getting below output after running the command (`docker run --rm -v "/home/renovate/renovate_config/config.js:/usr/src/app/config.js" renovate/renovate`)
```
INFO: DRY-RUN: Would commit files to branch renovate/major-springfox-version (repository=project/testrepo, branch=renovate/major-springfox-version)
 INFO: DRY-RUN: Would close Dependency Dashboard (repository=project/testrepo)
       "title": "Dependency Dashboard"
 INFO: Repository finished (repository=project/testrepo)
       "cloned": true,
       "durationMs": 1139308
```

**Reply by viceice** [reply]
you forgot to set debug log level as environment variable. read the getting started guide for self hosting.

**Reply by viceice** [reply]
you also set dry run, so renovate won't do any change on the repo. please read the docs.

**Reply by mikemicky4321** [reply]
Hello @viceice,
It worked perfectly after adjusting dry run. I have a question, We would like to use renovate on regular basis. We don't have to run this every time right? Once the docker was up and running using the command it work continuously  i.e., if anyone create a new repository and included renovate.json then they should receive PR with vulnerable libraries. right?

**Reply by viceice** [reply]
no,you need to run renovate hourly. the oss cli , which you are using will only run once

**Reply by mikemicky4321** [reply]
Okay. But I am running using `docker run -d -t -v "/home/renovate/renovate_config/config.js:/usr/src/app/config.js" renovate/renovate` to make the docker running continuously without exiting. If this doesn't work. Could you please help us to understand how to make the renovate to run every day automatically?
Thanks for your support

**Reply by viceice** [reply]
if you check those container, it will simply exit when done and remain stopped.

you need some external Cron job.

can be done Linux crontab or Jenkins pipeline or kubernetes cronjob ....

**Reply by mikemicky4321** [reply]
Okay. I have a question. Is it possible to use token based authentication in config.js? becuase when I have configured token it throws me the following error 'You must configure Bitbucket Server using Username/Password'. Is it true?

**Reply by viceice** [reply]
try to use the token as password

**Reply by mikemicky4321** [reply]
That's awesome. Thanks for your support.
